pkg/ebpftracer/decoder/decoder.go

```diff
@@ -612,13 +629,83 @@ func (decoder *Decoder) ReadArgsArrayFromBuff() ([]string, error) {
 	if ss[len(ss)-1] == "" {
 		ss = ss[:len(ss)-1]
 	}
+
+	fmt.Printf("\n\n decoder.redactSensitiveValues: %v\n", decoder.redactSensitiveValues)
+	if decoder.redactSensitiveValues {
+		decoder.redactArgs(ss)
+	}
+	fmt.Printf("\n\n args: %v\n", ss)
+
 	for int(argNum) > len(ss) {
 		ss = append(ss, "?")
 	}
 
 	return ss, nil
 }
 
+// redactArgs redacts the arguments that we consider sensitive, e.g. password, token, etc.
+//
+// It transforms this
+//
+//	["cmd", "--password", "abc123"]
+//
+// to this:
+//
+//	["cmd", "--password", "<redacted>"]
+func (decoder *Decoder) redactArgs(args []string) {
```

Comment:
This is not the best place to redact values. I would move it maybe to castai events exporter https://github.com/castai/kvisor/blob/main/cmd/agent/daemon/state/castai_events_exporter.go#L49

The reason is that I still want to get full real values then developing ebpf tracer.

Reply:
To be honest, I would move it to an enricher (even though it is technically not enriching, haha).